docs/content/tutorial_code/getting_started_1.md

```diff
@@ -1,7 +1,321 @@
+```go
+FilterHeadersStatus AddHeaderContext::onResponseHeaders(uint32_t) {
+  LOG_DEBUG(std::string("onResponseHeaders ") + std::to_string(id()));
+  addResponseHeader("newheader", root_->header_value_);
+  addResponseHeader("doc-header", "it-worked");
+  replaceResponseHeader("location", "envoy-wasm");
+  return FilterHeadersStatus::Continue;
+}
+```
+
+## Building the filter
+
+Now, let's build our filter. We can build using Bazel if we have that installed, or we can leverage a docker container that sets up all of the dependencies for us.
+
+For Bazel directly:
+
+```shell
+$  bazel build :filter.wasm
+```
+
+From the docker container using the `wasme` tooling:
+
+```shell
+$  wasme build . 
+```
+
+This will download all of the necessary dependencies and compile the output `filter.wasm` into the `_output` folder if using the `wasme` tooling. Otherwise you can find filter in `./bazel-bin/filter.wasm`.
+
+Now that we've built the `wasm` module, let's package it and load it into a registry so we can consume it in our Envoy proxy.
+
+## Pushing wasm module to registry
+
+Before we use our new filter.wasm module, let's push it into a registry that can be used as a source for the module when we configure Envoy. Alternatively we could try load it from the disk, but that involves a lot more.
+
+To do that, let's login to the `webassemblyhub.io` using GitHub as the OAuth provider. From the CLI:
+
+```shell
+$  wasm login
+
+Using port: 60632
+Opening browser for login. If the browser did not open for you, please go to:  https://webassemblyhub.io/authorize?port=60632
+```
+
+You should see a GitHub OAuth screen:
+
+![](/img/wasme_login.png)
+
+Click the "Authorize" button at the bottom and continue.
+
+After successful auth, you should see this in the terminal:
+
+```shell
+success ! you are now authenticated
+```
+
+Now let's push to the webassemblyhub.io registry. 
+
+```shell
+$  wasme push webassemblyhub.io/christian-posta/test:v0.1 ./_output_/filter.wasm
+```
+
+NOTE: The tag name to use is
+`webassemblyhub.io/<your-git-username>/<whatever-name>:<whatever-version>`
+
+When you've pushed, you should be able to see your new module in the registry:
+
+```shell
+$  wasme list
+
+NAME                        SHA      UPDATED             SIZE   TAGS
+christian-posta/test        6aef37f3 13 Jan 10 12:54 MST 1.0 MB v0.1
+```
+
+## Deploy our new module
+
+To deploy this new WASM module into Envoy, using Gloo we can configure the gateway to load the module from the `webassembly.io` registry:
+
+```yaml
+apiVersion: gateway.solo.io.v2/v2
+kind: Gateway
+metadata:
+  labels:
+    app: gloo
+  name: gateway-proxy-v2
+  namespace: gloo-system
+spec:
+  bindAddress: '::'
+  bindPort: 8080
+  httpGateway:
+    plugins:
+      extensions:
+        configs:
+          wasm:
+            image: webassemblyhub.io/christian-posta/test:v0.1
+            name: christian
+            root_id: add_header_root_id
+  proxyNames:
+  - gateway-proxy-v2
+  useProxyProto: false
```

Comment:
the master README.md has an updated version of what this should look like for the new version of gloo

Reply:
Okay, yah, I need to update for that.